added support for device detection

jlquaccia opened the pull request:
i referenced some device detection logic from here: https://github.com/matthewhudson/current-device (i've used current-device before in the past).  i also added device detection for tablet as well since it was easy to pluck from the library.

so with this pr, device detection covers mobile, tablet & desktop.

pm-harshad-mane commented:
please move the new code to a smaller function
Let's use "platform" key in the final JSON

jlquaccia commented:
Addressed the following in the commit I pushed today:

- now capturing network type and passing to worker
- wrapped entire onPage script in an IIFE
- keeping track of the sequence of order that bidder calls are made each auction
- keeping track of the number of ad units that are present during each auction
- keeping track of the request size (collecting the char count), for each bidder (per auction)
- keeping track of the bidder req call type (post, get, etc.) (per auction)
- made onPage script modular, can now dynamically decide which bidder calls to monitor based on who is participating in an auction
- ensuring that the cf worker only gets called once per auction
- tracking perceived latency (responseEnd - fetchStart)
- keeping track of bidder responses (per auction), whether or not they were used in pbjs
- keeping track of the number of calls a bidder makes (per auction)

pm-harshad-mane commented on OnPageCode.js:
```diff
@@ -1,196 +1,334 @@
+    // to make sure that we do not read the stats for the same network call again
+    'lastExecutionMaxIndex': 0,
+
+    'networkType': navigator.connection.effectiveType,
```
are we sure that the `navigator.connection.effectiveType,` is supported by all browsers? I feel we should handle the case where it is not supported by a browser.

jlquaccia replied:
good call to play it safe on this.  i can look into compatibility around `navigator.connection.effectiveType`

jlquaccia commented:
Just pushed up a new commit with the following changes:

- Modified calls to cf worker so they only happen once per bidder
- Changed `resUsedInPbjsAuction` value to boolean
- updated logic around capturing perceived latency (referred to logic done in pubmaticAnalyticsAdapter)
- Researched browser compatibility around `navigator.connection.effectiveType`.  spent about half a day looking for an alternative solution for browsers that don't support this, but was unable to find a clean solution.  `networkType` will now only be captured if a browser supports `navigator.connection` otherwise the `networkType` field will not be included in the payload
- Updated the logic around geo caching.  Created a new CF worker script that captures user geo data and returns to `onPageCode.js`.  the cf worker is called once every 24 hours, it's response is stored in localstorage in the browser.  the localstorage value will always be used unless it is a user's initial visit or 24 hours have passed since the last call to the worker.
- updated the payload to ELK to include the cf geo data
- moved the logic to post data to ELK into the `onPageCode.js` script
